src/wp-includes/class-wp-scripts.php

```diff
@@ -757,6 +757,103 @@ private function get_normalized_script_args( $handle, $args = array() ) {
 		return wp_parse_args( $args, $default_args );
 	}
 
+	/**
+	 * Get all of the scripts that depend on a script.
+	 *
+	 * @param string $handle The script handle.
+	 * @return array Array of script handles.
+	 */
+	private function get_dependents( $handle ) {
+		$dependents = array();
+
+		// Iterate over all registered scripts, finding ones that depend on the script.
+		foreach ( $this->registered as $registered_handle => $args ) {
+			if ( in_array( $handle, $args->deps, true ) ) {
+				$dependents[] = $registered_handle;
+			}
+		}
+		return $dependents;
+	}
+
+	/**
+	 * Get the strategy mentioned during script registration.
+	 *
+	 * @param string $handle The script handle.
+	 * @return string|bool Strategy set during script registration. False if none was set.
+	 */
+	private function get_intended_strategy( $handle ) {
+		$script_args = $this->get_data( $handle, 'script_args' );
+		return isset( $script_args['strategy'] ) ? $script_args['strategy'] : false;
+	}
+
+	/**
+	 * Check if all of a scripts dependents are deferrable which is required to maintain execution order.
+	 *
+	 * @param string $handle  The script handle.
+	 * @param array $visited An array of already visited script handles used to avoid looping recursion.
+	 * @return bool True if all dependents are deferrable, false otherwise.
+	 */
+	private function all_dependents_are_deferrable( $handle, $visited = array() ) {
```

Comment:
Readability reasons:

```suggestion
	/**
	 * Check if all of a scripts dependents are deferrable which is required to maintain execution order.
	 *
	 * @param string $handle  The script handle.
	 * @param array $checked An array of already visited script handles, used to avoid looping recursion.
	 * @return bool True if all dependents are deferrable, false otherwise.
	 */
	private function all_dependents_are_deferrable( $handle, $checked = array() ) {
```